### ISSUE TITLE
 Feature Request: Improved PF2E compatability

### ISSUE THREAD

**Issue by cadowtin** [Saibot393/LocknKey#8]
I have a possible path forward to improve compatibility with the pf2e system for **force open** and **pick a lock**.

You would simply need to call the following functions `game.pf2e.actions.pickALock` and `game.pf2e.actions.forceOpen` with parameters like so:
```javascript
game.pf2e.actions.pickALock(
    {
        callback: 
            async (roll) => {
                //Put all code in this callback related to picking/changing the lock status
                switch (roll.outcome) {
                    case 'criticalFailure':
                        break;
                    case 'failure':
                        break;
                    case 'success':
                        break;
                    case 'criticalSuccess':
                        break;
                    default:
                        //treat as fail
                        break;
                }
            }, 
    difficultyClass: 1 //set the lockpicking DC here
})
```

This integration would help a lot making the compatibility with the pf2e system as a whole, as it allows it to play nicely with pf2e's unique system concept of rule elements etc.

I would make the change myself except I believe you understand the inner workings of your code far better than I do upon my first look.

But yes work like this would be appreciated (but i understand if it's too much)

**Comment by Saibot393** [comment]
Thanks for letting me know. I will look into it, should be relatively easy to implement.

**Comment by Saibot393** [comment]
Both roll functions are now implemented with v1.6.0
